### PR TITLE
Support callback_url from processor

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Adds an optional `callback_url` field for the /api/results/upload endpoint, which dictates where we post the `TestRun` to. This is necessary when we're integration-testing a staging instance with modified behaviour of the `/api/results/create` endpoint (i.e. for the wpt.fyi checks).

### CODE REVIEW

File: api/taskcluster/webhook.go [hunk 1]
```diff
@@ -326,6 +337,9 @@ func createRun(client *http.Client, sha, api string, username string, password s
 	if labels != nil {
 		payload.Add("labels", strings.Join(labels, ","))
 	}
+	// Ensure we call back to this appengine version instance.
+	host := appengine.DefaultVersionHostname(ctx)
```
Comment (Hexcles) [inline api/taskcluster/webhook.go]: I don't think this does what your comment says: it's the *default* version, not this (the current) version.

Comment (Hexcles) [reply api/taskcluster/webhook.go]: What about `shared.GetHostname`?

Comment (lukebjerring) [reply api/taskcluster/webhook.go]: That's what I had intended to call, done.

File: shared/datastore.go [hunk 2]
```diff
@@ -1,14 +1,14 @@
 package shared
 
 import (
+	"context"
```
[the remark this answers is not in the document]
Comment (Hexcles) [reply shared/datastore.go]: These are good to have. `context` was experimental but has graduated to stable long ago; I always change `golang.org/x/net/context` to `context` whenever I see one.

**Pull request merged**